test/Riok.Mapperly.Tests/Mapping/StaticMethodConversionTest.cs

```diff
@@ -0,0 +1,3006 @@
+using Riok.Mapperly.Abstractions;
+using Riok.Mapperly.Diagnostics;
+
+namespace Riok.Mapperly.Tests.Mapping;
+
+public class StaticMethodConversionTest
+{
+    [
+        Theory,
+        InlineData("byte", "class  A { public static A Create(byte source) => new(); }", "return global::A.Create(source);"),
+        InlineData("byte", "struct A { public static A Create(byte source) => new(); }", "return global::A.Create(source);"),
+        InlineData("byte", "class  A { public static A CreateFrom(byte source) => new(); }", "return global::A.CreateFrom(source);"),
+        InlineData("byte", "struct A { public static A CreateFrom(byte source) => new(); }", "return global::A.CreateFrom(source);"),
+        InlineData(
+            "byte",
+            "class  A { public static A CreateFromByte(byte source) => new(); }",
+            "return global::A.CreateFromByte(source);"
+        ),
+        InlineData(
+            "byte",
+            "struct A { public static A CreateFromByte(byte source) => new(); }",
+            "return global::A.CreateFromByte(source);"
+        ),
+        InlineData("byte", "class  A { public static A FromByte(byte source) => new(); }", "return global::A.FromByte(source);"),
+        InlineData("byte", "struct A { public static A FromByte(byte source) => new(); }", "return global::A.FromByte(source);"),
+        InlineData("sbyte", "class  A { public static A Create(sbyte source) => new(); }", "return global::A.Create(source);"),
```

Comment:
This test suite seems overly extensive for the current purpose. In my opinion, a single representative test case for each scenario would be sufficient. For example:
* One test case per method name.
* One to handle keyword-specific cases.
* One for nullable and non-nullable value/reference types.
* One to verify method name case sensitivity.
This would ensure the key constellations are covered without unnecessary redundancy.